pkg/user/tx_client.go

```diff
@@ -673,3 +676,23 @@ func QueryNetworkMinGasPrice(ctx context.Context, grpcConn *grpc.ClientConn) (fl
 	}
 	return networkMinPrice, nil
 }
+
+// handleUnknownStatus handles transactions that return an unknown status.
+// It reverts the sequence number and removes the transaction from tracking.
+func (client *TxClient) handleUnknownStatus(txHash string) error {
```

Comment:
This is really similar to the handleEvictions function. Maybe we can combine the two and just return different errors.